src/plugins/discourse/mirror.test.js

```diff
@@ -868,4 +877,587 @@ describe("plugins/discourse/mirror", () => {
       });
     });
   });
+
+  describe("mirror mode 2", () => {
+    const example = exampleMode("MODE_2_LOAD_BUMPED_TOPICS");
+    it("mirrors topics from the fetcher", async () => {
+      const {mirror, fetcher, reporter, repo} = example();
+      const t1 = fetcher.addTopic();
+      const t2 = fetcher.addTopic();
+      await mirror.update(reporter);
+      expect(repo.topics()).toEqual([
+        fetcher._topic(t1.topicId),
+        fetcher._topic(t2.topicId),
+      ]);
+    });
+
+    it("mirrors category definition topics from the fetcher", async () => {
+      const {mirror, fetcher, reporter, repo} = example();
+      const c1 = fetcher.addCategory();
+      const t2 = fetcher.addTopic();
+      await mirror.update(reporter);
+      expect(repo.topics()).toEqual([
+        fetcher._topic(c1.topicId),
+        fetcher._topic(t2.topicId),
+      ]);
+    });
+
+    it("mirrors topics in a category from the fetcher", async () => {
+      const {mirror, fetcher, reporter, repo} = example();
+      const c1 = fetcher.addCategory();
+      const t2 = fetcher.addTopic({categoryId: c1.categoryId});
+      await mirror.update(reporter);
+      expect(repo.topics()).toEqual([
+        fetcher._topic(c1.topicId),
+        fetcher._topic(t2.topicId),
+      ]);
+    });
+
+    it("mirrors posts from the fetcher", async () => {
+      const {mirror, fetcher, reporter, repo} = example();
+      const t1 = fetcher.addTopic();
+      const t2 = fetcher.addTopic();
+      const [p3] = [fetcher.addPost({topicId: t2.topicId})];
+      await mirror.update(reporter);
+      const posts = [
+        fetcher._post(t1.postId),
+        fetcher._post(t2.postId),
+        fetcher._post(p3),
+      ];
+      expect(repo.posts()).toEqual(posts);
+    });
+
+    it("mirrors edited posts from the fetcher", async () => {
+      const {mirror, fetcher, reporter, repo} = example();
+      const t1 = fetcher.addTopic();
+      const t2 = fetcher.addTopic();
+      const p3 = fetcher.addPost({topicId: t2.topicId});
+      fetcher.editPost(p3, {cooked: "<p>Here's some edited content.</p>"});
+      await mirror.update(reporter);
```

Comment:
what would happen if the mirror updates, and then we edit the post, and then we update the mirror again? can we have a test for this case?

Reply:
This is covered, for example in "mirrors updates normal post edits, not category definition post edits".

https://github.com/sourcecred/sourcecred/blob/c12b148ae5b4805a0d084b754fb71f79678db7a8/src/plugins/discourse/mirror.test.js#L991-L1029